app/models/service_template.rb

```diff
@@ -74,6 +76,7 @@ class ServiceTemplate < ApplicationRecord
   scope :without_service_template_catalog_id,       ->         { where(:service_template_catalog_id => nil) }
   scope :with_existent_service_template_catalog_id, ->         { where.not(:service_template_catalog_id => nil) }
   scope :displayed,                                 ->         { where(:display => true) }
+  scope :public_service_templates,                  ->         { where.not(:id => ServiceTemplate.includes(:reserved_rec).joins(:reserved_rec).select { |s| s.internal == true }.collect(&:id)) }
```

Comment:
not sure if the `joins` is necessary

Reply:
@kbrock What if we query the Reserve first?
```
where.not(:id => Reserved.where(:resource_type => "ServiceTemplate").all.collect { |r| r.resource_id if r[:internal] == true }.compact)
```

Reply:
@kbrock Without `join`, `ServiceTemplate.includes(:reserved_rec).select { |s| s.internal == true }.collect(&:id)` give us same result.

Reply:
@hsong-rh Remember we need to retain this as a scope so users can chain other scopes onto the resulting object.

Reply:
@bdunne This works: `where.not(:id => Reserve.where(:resource_type => "ServiceTemplate").all.collect { |r| r.resource_id if r.reserved[:internal] == true }.compact)

Reply:
@gmcculloug yes, they return as scope.